staging/src/k8s.io/cloud-provider/cloud.go

```diff
@@ -186,6 +186,7 @@ type Instances interface {
 	// This method should still return true for instances that exist but are stopped/sleeping.
 	InstanceExistsByProviderID(ctx context.Context, providerID string) (bool, error)
 	// InstanceShutdownByProviderID returns true if the instance is shutdown in cloudprovider
+	// If the instance is deleted, this method should return false, or throw an error.
```

Comment:
I don't like 'or'. 
I think we should suggest the boolean behavior. 
Error should be an indicator that something went wrong and we could not determine the answer. (Eg. failed to connect to the relevant service or the function has not been implemented)
Also I believe if the instance has been deleted, then it has been shutdown and we should be returning true.

Reply:
@cheftako returning true completely ignores the delete in the controller, and the nodes is never deleted :sweat_smile: a `cloudprovider.InstanceNotFound` makes sense for me though (more than `false` anyway` :smile: )

Reply:
(Today returning false or an error have the same behaviour, except the error is logged)